docs/pipelines.md

```diff
@@ -553,10 +550,177 @@ spec:
         Name: cleanup
 ```
 
-_[PR #2661](https://github.com/tektoncd/pipeline/pull/2661) is implementing this new functionality by adding support to enable
-final tasks along with workspaces and parameters. `PipelineRun` status is being updated to include execution status of
-final tasks i.e. `PipelineRun` status is set to success or failure depending on execution of `PipelineTasks`, this status
-remains same when all final tasks finishes successfully but is set to failure if any of the final tasks fail._
+### Specifying `Workspaces` in Final Tasks
+
+Finally tasks can specify [workspaces](workspaces.md) which `PipelineTasks` might have utilized
+e.g. a mount point for credentials held in Secrets. To support that requirement, you can specify one or more
+`Workspaces` in the `workspaces` field for the final tasks similar to `tasks`.
+
+```yaml
+spec:
+  resources:
+    - name: app-git
+      type: git
+  workspaces:
+    - name: shared-workspace
+  tasks:
+    - name: clone-app-source
+      taskRef:
+        name: clone-app-repo-to-workspace
+      workspaces:
+        - name: shared-workspace
+          workspace: shared-workspace
+      resources:
+        inputs:
+          - name: app-git
+            resource: app-git
+  finally:
+    - name: cleanup-workspace
+      taskRef:
+        name: cleanup-workspace
+      workspaces:
+        - name: shared-workspace
+          workspace: shared-workspace
+```
+
+### Specifying `Parameters` in Final Tasks
+
+Similar to `tasks`, you can specify [`Parameters`](tasks.md#specifying-parameters) in final tasks:
+
+```yaml
+spec:
+  tasks:
+    - name: tests
+      taskRef:
+        Name: integration-test
+  finally:
+    - name: report-results
+      taskRef:
+        Name: report-results
+      params:
+        - name: url
+          value: "someURL"
+```
+
+### `PipelineRun` Status with `finally`
+
+With `finally`, `PipelineRun` status is calculated based on `PipelineTasks` under `tasks` section and final tasks.
+
+Without `finally`:
+
+| `PipelineTasks` under `tasks` | `PipelineRun` status | Reason |
+| ----------------------------- | -------------------- | ------ |
+| all `PipelineTasks` successful | `true` | `Succeeded` |
+| one or more `PipelineTasks` skipped and rest successful | `true` | `Completed` |
```

Comment:
I have included a link to conditions doc, but still not clear, we need a separate doc for states and explaining them in detail 😿

Reply:
`skipped` is overly used, task is skipped if a condition fails for a conditional task, task is skipped if any one of the parent skipped, task is skipped if a task is never scheduled because of other task failure (pipeline exits)

Reply:
🤦‍♀️  looks like I added the link in the row [below](https://github.com/tektoncd/pipeline/pull/2661/files#diff-db9ac78ce08367ea2b7db8c1412049b6R623) instead of this row, added here as well now 😜